feat: [DDS-62/DDS-85] Embedded verification relationships and more tests

spivachuk opened the pull request:
- Added support for embedded verification relationships.
- Added more tests for verification relationships.

spivachuk committed:
[DDS-62] Support embedded verification relationships

- Updated proto files.

spivachuk committed:
[DDS-62] Support embedded verification relationships

- Regenerated *.pb.go files.

spivachuk committed:
[DDS-62] Support embedded verification relationships

- Fixed compilation errors after *.pb.go files regeneration.

spivachuk committed:
[DDS-62] Support embedded verification relationships

- Updated unit tests correspondingly to the recent changes.

spivachuk committed:
[DDS-62] Support embedded verification relationships

- Enhanced validation of verification relationships.
- Corrected unit tests.

Toktar committed:
[DDS-62]: Add unit tests for Verification Relationships

Toktar committed:
[DDS-62]: fix linter

spivachuk committed:
[DDS-62] Support embedded verification relationships

- Added more unit tests.

spivachuk committed:
[DDS-62] Support embedded verification relationships

- Updated keeper.VerifySignature function correspondingly to the new DidDoc structure.

spivachuk committed:
[DDS-62] Support embedded verification relationships

- Added unit tests verifying the changes in keeper.VerifySignature function.
- Corrected a comment.

spivachuk committed:
[DDS-62] Support embedded verification relationships

- Added integration tests verifying the changes in keeper.VerifySignature function.
- Corrected the updates made in existing integration tests.

spivachuk committed:
[DDS-85] DID update in verification relationships

- Added DID replacement in verification relationships to DidDoc.ReplaceDids method.

spivachuk committed:
[DDS-85] Corrected controllers retrieval logic

- Corrected controllers retrieval logic correspondingly to the latest changes.

spivachuk committed:
[DDS-85] Disabled SC2086 shellcheck linter warning

spivachuk committed:
[DDS-85] Added unit tests for verification relationships

Toktar commented on x/did/types/fee.pb.go:
```diff
@@ -81,30 +81,31 @@ func init() {
-	0x53, 0x04, 0x9e, 0x76, 0x48, 0xdb, 0xec, 0x90, 0xf6, 0xbe, 0x43, 0xda, 0xfd, 0xf9, 0xbe, 0x6d,
-	0xb9, 0x3d, 0x79, 0xb6, 0x02, 0xe6, 0x52, 0xb2, 0x92, 0xab, 0x94, 0xde, 0xed, 0xbf, 0x32, 0xeb,
-	0x8b, 0xef, 0x00, 0x00, 0x00, 0xff, 0xff, 0x20, 0xf3, 0xe8, 0x95, 0xe4, 0x01, 0x00, 0x00,
+	// 374 bytes of a gzipped FileDescriptorProto
```
Why do we have changes in this file?

Toktar commented on x/did/types/genesis.pb.go:
```diff
@@ -142,28 +142,28 @@ func init() {
-	0x20, 0xcc, 0xf4, 0x24, 0x72, 0x2d, 0xdb, 0x79, 0x94, 0x6b, 0xc0, 0xe8, 0xc9, 0x56, 0x4e, 0xcf,
-	0x05, 0x9a, 0xc1, 0x7f, 0x5b, 0xf9, 0xf1, 0x67, 0x00, 0x00, 0x00, 0xff, 0xff, 0xed, 0xeb, 0xc7,
-	0x91, 0xd1, 0x01, 0x00, 0x00,
+	// 328 bytes of a gzipped FileDescriptorProto
```
Why do we have changes in this file?

Toktar commented on x/did/types/query.pb.go:
```diff
@@ -318,35 +318,36 @@ func init() {
-	0x97, 0x5e, 0xd7, 0xc7, 0x4c, 0x4e, 0x92, 0x91, 0x1b, 0xf0, 0xa9, 0xf6, 0x51, 0xd7, 0x76, 0xc4,
-	0x09, 0xc5, 0x6f, 0x95, 0xa9, 0x5c, 0xc4, 0x54, 0x8c, 0x2e, 0xd4, 0x1f, 0xde, 0xf9, 0x19, 0x00,
-	0x00, 0xff, 0xff, 0x99, 0xb4, 0x16, 0xb9, 0x44, 0x04, 0x00, 0x00,
+	// 449 bytes of a gzipped FileDescriptorProto
```
Why do we have changes in this file?

Toktar commented on x/did/types/v1/common.pb.go:
```diff
@@ -81,19 +81,19 @@ func init() {
 func init() { proto.RegisterFile("cheqd/did/v1/common.proto", fileDescriptor_488ad3ab822f2393) }
 
 var fileDescriptor_488ad3ab822f2393 = []byte{
-	// 178 bytes of a gzipped FileDescriptorProto
```
Why do we have changes in v1 files?

Toktar reviewed (commented):
Thanks for a great job!
Just a few questions:
- Can we remove changes in proto generated files where it's not needed?
- Does this PR contain changes for DID Resource validation?
- Do we have a unit test where VerificationRelationship contains both ID and embedded VM (invalid case)?
- Could you please make a merge with `develop` brunch?

spivachuk committed:
[DDS-85] Added more tests of resource creation

- Added integration tests of resource creation using signatures by different verification methods / relationships.

spivachuk committed:
Merge branch 'develop' of github.com:canow-co/cheqd-node into did-doc-integration-tests-improvement

# Conflicts:
#	go.mod
#	go.sum
#	x/did/types/diddoc_diddoc.go
#	x/did/types/fee.pb.go
#	x/did/types/query.pb.go
#	x/did/types/v1/did.pb.go
#	x/did/types/v1/query.pb.go
#	x/did/types/v1/tx.pb.go
#	x/resource/types/fee.pb.go
#	x/resource/types/genesis.pb.go
#	x/resource/types/query.pb.go
#	x/resource/types/resource.pb.go
#	x/resource/types/v1/query.pb.go

spivachuk committed:
Resolved conflicts after merge

spivachuk committed:
Resolved conflicts after merge

spivachuk commented:
@Toktar Below please find answers to the questions from your previous comment:

>Can we remove changes in proto generated files where it's not needed?

Agreed to keep changes in these files "as is".

>Does this PR contain changes for DID Resource validation?

It contains changes in `DidDoc` validation. No changes are needed in `Resource` validation.

>Do we have a unit test where VerificationRelationship contains both ID and embedded VM (invalid case)?

Yes, it is [here](https://github.com/canow-co/cheqd-node/blob/877e5be754b022dd4218260e9efe2661a3604f67/x/did/types/diddoc_verification_relationship_test.go#L74).

>Could you please make a merge with `develop` brunch?

Done.
